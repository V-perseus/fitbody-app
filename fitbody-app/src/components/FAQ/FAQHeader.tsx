import React from 'react'
import { ScrollView } from 'react-native'
import { ActiveTab } from '../../screens/tabs/account/FAQ/FAQ'
import FAQButton from './FAQButton'

interface IFAQHeaderProps {
  active: ActiveTab
  handlePress: (tab: ActiveTab) => void
}
const FAQHeader: React.FC<IFAQHeaderProps> = ({ active, handlePress }) => {
  return (
    <ScrollView horizontal={true}>
      <FAQButton active={active === 'support'} title="TECH SUPPORT" handlePress={() => handlePress('support')} />
      <FAQButton active={active === 'workouts'} title="WORKOUTS" handlePress={() => handlePress('workouts')} />
      <FAQButton active={active === 'meals'} title="MEALS" handlePress={() => handlePress('meals')} />
      <FAQButton active={active === 'community'} title="COMMUNITY" handlePress={() => handlePress('community')} />
    </ScrollView>
  )
}

export default FAQHeader
