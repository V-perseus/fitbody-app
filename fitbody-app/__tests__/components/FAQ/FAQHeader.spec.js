import React from 'react'
import { cleanup, render } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import FAQHeader from '../../../src/components/FAQ/FAQHeader'

describe('FAQHeader', () => {
  const availableTabs = ['support', 'workouts', 'meals', 'community']

  let handlePress
  beforeEach(() => {
    handlePress = jest.fn()
  })

  afterEach(cleanup)

  it('renders snapshot correctly', () => {
    const tree = renderer.create(<FAQHeader active={availableTabs[0]} handlePress={handlePress} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correct tabs', () => {
    const { queryByText } = render(<FAQHeader active={availableTabs[0]} handlePress={handlePress} />)
    expect(queryByText(/tech support/i)).not.toBeNull()
    expect(queryByText(/workouts/i)).not.toBeNull()
    expect(queryByText(/meals/i)).not.toBeNull()
    expect(queryByText(/community/i)).not.toBeNull()
  })
})
