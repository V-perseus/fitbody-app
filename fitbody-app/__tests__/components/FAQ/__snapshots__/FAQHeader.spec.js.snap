// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQHeader renders snapshot correctly 1`] = `
<RCTScrollView
  horizontal={true}
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "paddingTop": 100,
        }
      }
      testID="tab_btn"
    >
      <BVLinearGradient
        colors={
          [
            4279220154,
            4286300671,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "alignItems": "center",
            "borderRadius": 50,
            "height": 57,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 9,
            "marginRight": 9,
            "marginTop": 5,
            "width": 160,
          }
        }
        testID="tab_bg"
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 16,
            }
          }
        >
          TECH SUPPORT
        </Text>
      </BVLinearGradient>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "paddingTop": 100,
        }
      }
      testID="tab_btn"
    >
      <BVLinearGradient
        colors={
          [
            4292796126,
            4292796126,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "alignItems": "center",
            "borderRadius": 50,
            "height": 45,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 9,
            "marginRight": 9,
            "marginTop": 10,
            "width": 140,
          }
        }
        testID="tab_bg"
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 16,
            }
          }
        >
          WORKOUTS
        </Text>
      </BVLinearGradient>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "paddingTop": 100,
        }
      }
      testID="tab_btn"
    >
      <BVLinearGradient
        colors={
          [
            4292796126,
            4292796126,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "alignItems": "center",
            "borderRadius": 50,
            "height": 45,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 9,
            "marginRight": 9,
            "marginTop": 10,
            "width": 140,
          }
        }
        testID="tab_bg"
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 16,
            }
          }
        >
          MEALS
        </Text>
      </BVLinearGradient>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "paddingTop": 100,
        }
      }
      testID="tab_btn"
    >
      <BVLinearGradient
        colors={
          [
            4292796126,
            4292796126,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "alignItems": "center",
            "borderRadius": 50,
            "height": 45,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 9,
            "marginRight": 9,
            "marginTop": 10,
            "width": 140,
          }
        }
        testID="tab_bg"
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 16,
            }
          }
        >
          COMMUNITY
        </Text>
      </BVLinearGradient>
    </View>
  </View>
</RCTScrollView>
`;
